src/hyp_map.rs

```diff
@@ -323,6 +325,63 @@ impl UmodeInputRegion {
     }
 }
 
+/// Represents a hypervisor stack region.
+pub struct HypStackRegion {
+    vaddr: PageAddr<SupervisorVirt>,
+    paddr: PageAddr<SupervisorPhys>,
+    page_count: u64,
+}
+
+impl HypStackRegion {
+    fn new(stack_pages: SequentialPages<InternalDirty>) -> Result<Self, Error> {
+        let page_count = stack_pages.len();
+        let paddr = stack_pages.base();
+        let vaddr = HYP_STACKTOP_PAGE_ADDR
+            .checked_sub_pages(page_count)
+            .ok_or(Error::InvalidStackSize(page_count))?;
+        if vaddr.bits() < HYP_STACKBTM_MIN {
+            return Err(Error::InvalidStackSize(page_count));
+        }
+        Ok(HypStackRegion {
+            vaddr,
+            paddr,
+            page_count,
+        })
+    }
+
+    // Map hypervisor stack in current page-table.
+    fn map(&self, sv48: &FirstStagePageTable<Sv48>, hyp_mem: &mut HypPageAlloc) {
+        let pte_fields = PteFieldBits::leaf_with_perms(PteLeafPerms::RW);
+        // Unmap stack pages from the 1:1 map.
+        for _ in sv48
+            .unmap_range(
+                self.paddr.as_supervisor_virt(),
+                PageSize::Size4k,
+                self.page_count,
+            )
+            .unwrap()
+        {}
```

Comment:
instead of an empty loop, you can use `count()` to consume the iterator.

Reply:
Ack.